cmd/bclient/fileutil/fileutil.go

```diff
@@ -0,0 +1,154 @@
+package fileutil
+
+// A catchall package for bclient file operations and data structures.
+// May need decomposition if too unwieldy
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+
+	"github.com/antonholmquist/jason"
+)
+
+var (
+	UpLoadDone     sync.WaitGroup
+	UseRemoteList  bool = true
+	localFileList  *FileList
+	remoteFileList *FileList
+	FilesWalked    = make(chan string)
+	rootPrefix     string
+	verbose        bool
+)
+
+//A public Method to get the Md5 sum of file on the upload list
+
+func ShowUploadFileMd5(fileName string) []byte {
+
+	return localFileList.Files[fileName][1]
+}
+
+// Print out remote filer list
+func PrintRemoteList() {
+	for key, value := range remoteFileList.Files {
+		fmt.Println("Key:", key, "Value:", value)
+	}
+
+}
+
+// Print out local file list
+func PrintLocalList() {
+
+	for key, value := range localFileList.Files {
+		fmt.Println("Key:", key, "Value:", value[1])
+	}
+
+}
+
+// Public Synchronization Gate
+
+func WaitFileListStep() {
+	IfVerbose("At UpLoadDone.Wait()")
+	UpLoadDone.Wait()
```

Comment:
moved the synch Group up into main, rolled the goroutines into anon functs, and delted this method and its ilk in fileutil